### ISSUE TITLE
Unable to use ExcelProvider on VisualStudio for Mac (Mono framework)

### ISSUE THREAD

**Issue by gaurav-clearTax** [fsprojects/ExcelProvider#33]
### Description

Program.fs(22,22): Error FS3033: The type provider 'ExcelProvider.ExcelProvider+ExcelProvider' reported an error: Could not create ExcelDataReader. Exception - Could not create ExcelDataReader. Exception - Method 'ICSharpCode.SharpZipLib.Zip.ZipEntry.IsCompressionMethodSupported' not found. (FS3033)

### Repro steps

1. Make a new project
2. Add a template xlsx file
3. Try using the ExcelProvider using `type TemplateExcel = ExcelFile<"TemplateType.xlsx">
`
4. Read the error messages

### Expected behavior
- Should be able to build

### Actual behavior
- Error while building

### Known workarounds
None that I could find out
Tried various versions of ExcelProvider ranging from 0.1.2 to 0.8.0
Please provide a description of any known workarounds.

### Related information 
* Operating system - Macintosh
* .NET Runtime, CoreCLR or Mono Version - Visual Studio for Mac (Mono framework)

**Comment by forki** [comment]
Yeah I guess someone should replace the ICSharpCode with something that runs on mono.

==> up-for-grabs

**Comment by gaurav-clearTax** [comment]
@forki Isn't the ICSharpCode a dependency for ExcelDataReader? and the loading of assemblies gives us the above error?

I was trying to fix this.. Do you have some work around in mind?

**Comment by forki** [comment]
Yeah I guess the bug is in ExcelDataReader and should be fixed over there. And then we just paket update the nupgk.

**Comment by gaurav-clearTax** [comment]
@forki 
I tried using ExcelDataReader and ICSharpCode.SharpZipLib independently in Visual Studio for Mac and for me it seems to work.

ICSharpCode.SharpZipLib is also being used in ExcelProvider for generating loadedAssemblies? I don't have much context on why we are doing that... Can you give me a bit of primer on this?

**Comment by forki** [comment]
I'm not the original author of this. so no sorry. All I can say is that lib needs to be replaced. But personally I don't have time for that

**Comment by randomizedthinking** [comment]
Encountered the same issue on Linux.

**Comment by quintusm** [comment]
Can anybody comment if this is still an issue with version2.0.0.

I don't have access to a Mac so can't confirm that it is sorted.
